File: Testes/T1/desconhecido/1_3.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

double f(double x){return exp(0.7*x)-pow(x,2)-0.5;};

int main() {
    double a=-1,b=0,m=(double)(a+b)/2;
    cout << a << " " << b << " " << m << " " << f(a) << " " << f(b) << " " << f(m) << endl;

    for(int i=0;i<2;i++){

        if (f(a)*f(m)<0)
            b=m;
        else
            a=m;
        m=(double)(a+b)/2;
        cout << a << " " << b << " " << m << " " << f(a) << " " << f(b) << " " << f(m) << endl;

    }

}

File: Testes/T1/desconhecido/1_3.py
import math

def f(x):
    return math.exp(0.7*x)-x**2-0.5

a=-1.0
b=0.0
i=0

while i<3:
    m=(a+b)/2
    print(str(a)+" " + str(b)+ " " + str(m)+" " + str(f(a))+" "+ str(f(b))+" "+ str(f(m)))
    if f(a)*f(m)<0:
        b=m
    else:
        a=m
    i=i+1

